mlforecast/distributed/forecast.py

```diff
@@ -208,6 +217,8 @@ def _preprocess_partitions(
             schema="ts:binary,train:binary,valid:binary",
             engine=self.engine,
             as_fugue=True,
+            partition=id_col,
```

Comment:
I don't support setting this unless we also set `num`, which should be a multiple of the number of machines. The reason is that we'll have as many tasks as series, which are probably over tens of thousands, which will introduce a lot of overhead and will use more memory since we'll have tens of thousands of objects being serialized.
Dask partitions based on the index and spark provides repartitionByRange, but I don't see something similar for ray.

Reply:
Yes, I found the problem with Ray where using only `repartition(num_partitions)` produces strange results (breaks certain series). I researched a bit and couldn't find a native way to create partitions based on a column and ended up hardcoding the `by` argument. Using fugue, I integrated the `num_partitions` argument in the newest commit to creating partitions based on `id_col`. If this argument is None, the original partitions of Dask and Spark will be respected; in that case, the partitions will be done by `id_col` for Ray. If `num_partitions` is used, the partitions will be overwritten. I included several tests to ensure that `partition_results` object contains the expected number of rows and that the number of partitions is correct for Spark, Dask, and Ray. I also included a test to show that the same metadata of the dataframes is returned before and after using `num_partitions`. With this change, for example, in Dask, it is no longer necessary to partition the dataframe by `id_col`. Let me know what you think :)

Reply:
I think that's a great approach! Just left a minor comment on the algo option.